File: leetcode/128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> m;
        for(auto& num: nums){
            m[num] = 1;
        }
        
        int res = 0;
        for(auto it = m.begin(); it != m.end(); ++it){
            int cnt = 1;
            int next = it->first + 1;
            while(true){
                auto next_it = m.find(next);
                if(next_it == m.end()){
                    break;
                }
                cnt += m[next];
                m.erase(next_it);
                ++next;
            }
            it->second = cnt;
            res = max(res, cnt);
        }
        return res;
    }
};

File: leetcode/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for i in range(len(nums)):
            if nums[i]+1 in s:
                continue
            curr = nums[i]
            cnt = 0
            while curr in s:
                s.remove(curr)
                curr -= 1
                cnt += 1
            res = max(res, cnt)
        return res


